seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/JdbcSource.java

```diff
@@ -236,6 +237,6 @@ private PartitionParameter initPartitionParameterAndExtendSql(Connection connect
     }
 
     private boolean isNumericType(SeaTunnelDataType<?> type) {
-        return type.equals(BasicType.INT_TYPE) || type.equals(BasicType.LONG_TYPE);
+        return type.equals(BasicType.INT_TYPE) || type.equals(BasicType.LONG_TYPE) || type instanceof DecimalType;
```

Comment:
> DecimalType.equals(type) and decimal may cause some exception for spilt, because how solve the float data?

I agree with you. The `max` and `min` type is long. It may case exception. 

```
long max = Long.MAX_VALUE;
        long min = Long.MIN_VALUE;
        if (jdbcSourceConfig.getPartitionLowerBound().isPresent()
                && jdbcSourceConfig.getPartitionUpperBound().isPresent()) {
            max = jdbcSourceConfig.getPartitionUpperBound().get();
            min = jdbcSourceConfig.getPartitionLowerBound().get();
            return new PartitionParameter(
                    columnName, min, max, jdbcSourceConfig.getPartitionNumber().orElse(null));
        }
        try (ResultSet rs =
                connection
                        .createStatement()
                        .executeQuery(
                                String.format(
                                        "SELECT MAX(%s),MIN(%s) " + "FROM (%s) tt",
                                        columnName, columnName, query))) {
            if (rs.next()) {
                max =
                        jdbcSourceConfig.getPartitionUpperBound().isPresent()
                                ? jdbcSourceConfig.getPartitionUpperBound().get()
                                : Long.parseLong(rs.getString(1));
                min =
                        jdbcSourceConfig.getPartitionLowerBound().isPresent()
                                ? jdbcSourceConfig.getPartitionLowerBound().get()
                                : Long.parseLong(rs.getString(2));
            }
        }

```

Reply:
but in most case, like in Oracle, the default number type's precision is 38， which will map to Decimal type in seatunnel. It should be better to throw exception when init split param than reject split in the first time. isNumericType will also throw exception, but it is much earlier.